Sources/NeedleFoundation/Internal/DependencyProviderRegistry.swift

```diff
@@ -26,6 +26,8 @@ import Foundation
 // method in the base component class. Generate extensions to all the
 // component subclasses that override the method to instantiate the
 // dependnecy providers.
+@preconcurrency
+@MainActor
 public class __DependencyProviderRegistry {
 
     /// The singleton instance.
```

Comment:
![Screenshot 2024-03-09 at 10 39 22 PM](https://github.com/uber/needle/assets/22668017/46c94f35-a383-4132-a91a-c424a1147da7)

```suggestion
    /// The singleton instance.
    @preconcurrency
    @MainActor
```

Reply:
@garricn May you please incorporate this suggestion into the PR? Thank you!

Reply:
Thank you! 👍